### PR TITLE
Add support for Rails 6

### PR DESCRIPTION
**Pull request by MatthiasRMS**
So this is the PR related to the ongoing [discussion](https://github.com/gonzalo-bulnes/simple_token_authentication/issues/346) about making this library compatible with Rails 6.

@gonzalo-bulnes I've changed the restrictions. You asked to write a spec about the Rails 6 compatibility. Where should I do this ? I couldn't find the specs for other versions in the `spec` folder besides the different gemfiles. I've added a gemfile for this version which works with [appraisal](https://github.com/thoughtbot/appraisal). 

I haven't done much so far, some directions on where to go from here would be greatly appreciated 😉

### CODE REVIEW

File: Appraisals [hunk 1]
```diff
@@ -1,3 +1,7 @@
+appraise 'rails_6_devise_4' do
+  # use gemspec constraints
```
Comment (gonzalo-bulnes) [inline Appraisals]: This is good. That's what I meant by adding test coverage for Rails 6. :+1:

File: Appraisals [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+end
+
 appraise 'rails_5_devise_4' do
   # use gemspec constraints
```
Comment (gonzalo-bulnes) [inline Appraisals]: The gemspec constraints will result in Rails 6 being installed now. So this combination needs to be constrained if we want it to actually test Rails 5. Latest Devise is still in the 4 series.

```suggestion
  gem 'actionmailer', '~> 5.0'
  gem 'actionpack', '~> 5.0'
  gem 'activerecord', '~> 5.0'
```

File: Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,148 @@
+PATH
```
Comment (gonzalo-bulnes) [inline Gemfile.lock]: Let's [not commit this `Gemfile.lock`](https://github.com/gonzalo-bulnes/simple_token_authentication/pull/324).

File: Appraisals (hunk 2 above)
Comment (gonzalo-bulnes) [reply Appraisals]: Note: the Gemfile generated by Appraisal (`gemfiles/rails_5_devise_4.gemfile`) will change as a result of this, those changes need to be committed together with this change.

File: gemfiles/rails_4_devise_3.gemfile [hunk 4]
```diff
@@ -7,4 +7,4 @@ gem "actionpack", "~> 4.0"
 gem "activerecord", "~> 4.0"
 gem "devise", "~> 3.2"
 
-gemspec :path => "../"
```
Comment (gonzalo-bulnes) [inline gemfiles/rails_4_devise_3.gemfile]: If this was done by Appraisal, all good. :+1: If you edited them manually, let not do it. These are generated files, and it is better to not ever edit them manually. :slightly_smiling_face:

File: gemfiles/rails_6_devise_4.gemfile [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# This file was generated by Appraisal
+
+source "https://rubygems.org"
+
+gemspec path: "../"
```
Comment (gonzalo-bulnes) [inline gemfiles/rails_6_devise_4.gemfile]: Yep, looks like Appraisal updated their format. :+1:

File: simple_token_authentication.gemspec [hunk 6]
```diff
@@ -16,13 +16,13 @@ Gem::Specification.new do |s|
   s.files = Dir["{app,config,doc,lib}/**/*", "CHANGELOG.md", "LICENSE", "Rakefile", "README.md"]
   s.test_files = Dir["spec/**/*", "gemfiles/*.gemfile", "gemfiles/*.gemfile.lock", "Appraisals"]
 
-  s.add_dependency "actionmailer", ">= 3.2.6", "< 6"
-  s.add_dependency "actionpack", ">= 3.2.6", "< 6"
+  s.add_dependency "actionmailer", ">= 3.2.6", "< 7"
```
Comment (gonzalo-bulnes) [inline simple_token_authentication.gemspec]: :+1: for all three constraints.

**Comment by gonzalo-bulnes** [comment]
Also, you likely want to update the [`CHANGELOG`](). I'd put it in the **Unreleased** section (there is a example in https://keepachangelog.com/en/1.0.0). That section can them be turned into **1.16.0** when buidling the release.

**Comment by MatthiasRMS** [comment]
Hi @gonzalo-bulnes ,

I've updated the changelog, let me know if this is sufficient.
As for checking deprecations messages, how should I do this ? How can I list the deprecations messages for Rails 6 ? I don't know an easy way to do this 😉

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased] - 2019-07-12
+
+- Add Rails 6 support
+- Lock gems for Rails 5 specs
+
 ## [Unreleased] - 2017-12-21
```
Comment (gonzalo-bulnes) [inline CHANGELOG.md]: No need to duplicate the header.

```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased] - 2019-07-12
+
```
Comment (gonzalo-bulnes) [inline CHANGELOG.md]: Also, the format of the change log (see https://keepachangelog.com/en/1.0.0) requires to classify the changes by category, in this case these qualify as addtiions.

```suggestion

### Added

```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,11 @@
...
+- Add Rails 6 support
+- Lock gems for Rails 5 specs
+
 ## [Unreleased] - 2017-12-21
 
 ## Fixed
```
Comment (gonzalo-bulnes) [inline CHANGELOG.md]: Oops, I just noticed my mistake here, it should be a third-level header.

```suggestion
### Fixed
```

Comment (gonzalo-bulnes) [reply CHANGELOG.md]: And same thing in the 1.15.1. section that I can't reach in this PR review, but if you feel like fixing it, please go ahead!

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased] - 2019-07-12
+
+- Add Rails 6 support
+- Lock gems for Rails 5 specs
```
Comment (gonzalo-bulnes) [inline CHANGELOG.md]: I don't think this is useful to know from a Simple Token Authentication user point of view. It is very much a gem maintenance concern.

```suggestion
```

**Comment by nickcoyne** [comment]
What do we need to do to get these builds passing? Rails 6 has been released.

**Comment by gonzalo-bulnes** [comment]
Not much @nickcoyne, namely [the three last commits in this branch](https://github.com/gonzalo-bulnes/simple_token_authentication/compare/spike-yard-constraints-for-ruby-1.9.3).

@MatthiasRMS Since I cannot add the commits to the branch in your fork, I copied over your commits (also grouped a few changes and reworded commit messages accordingly) to be able to get a green build and ship this. (:tada:)

**Comment by gonzalo-bulnes** [comment]
So many thanks @MatthiasRMS! :confetti_ball: You can now pull your code with the freshly released **v1.16.0** :wink:

**Pull request closed without merging**